packages/data/lib/src/request/trakt_api/trakt_api_movies.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:data/src/request/api_request_representable.dart';
+import 'package:data/src/request/trakt_api/constants/headers.dart';
+import 'package:data/src/request/trakt_api/constants/query.dart';
+import 'package:data/src/request/trakt_api/extensions/movies_ext.dart';
+import 'package:domain/src/json/yaml/properties.dart';
+
+enum MoviesEndpoint { trending, anticipated }
```

Comment:
about this enum, it seems to be redundant. why? let's imagine the situation, we have 2 films data sources (like trakt and imdb). we will create 2 repositories for them. this repositories will be passed to usecase and we will have the situation like this:

```
class GetTrendingListUseCase {
  TraktRepository traktRepo;
  IMDBRepository imdbRepo;

  Future<data> call() {
      if (something) return traktRepo.getTrendingList();
      else  return imdbRepo.getTrendingList();
}
```

so, as u can see there's no need in the enum in this case

Reply:
This enum serves for obtaining endpoint inside the traktApi:
endPoint.string; =>
'/movies/trending'    ||     '/movies/anticipated'
in this case it is not full endpoint but only part of it. With out him I will be forced to write almost same request class.